synapse/storage/util/id_generators.py

```diff
@@ -352,12 +356,39 @@ def _mark_id_as_finished(self, next_id: int):
 
         with self._lock:
```

Comment:
Unrelated, but I just noticed that "Position" is spelled incorrect in the docstring. 😄

Reply:
Ta!